analytic_engine/src/row_iter/record_batch_stream.rs

```diff
@@ -77,83 +101,76 @@ pub type SequencedRecordBatchStream = Box<
         + Unpin,
 >;
 
-/// Filter the `sequenced_record_batch` according to the `filter` if necessary.
-/// Returns the original batch if only a small proportion of the batch is
-/// filtered out.
-/// The `selected_rows_buf` is for reuse.
+/// Filter the `sequenced_record_batch` according to the `predicate` if
+/// necessary. Returns the original batch if only a small proportion of the
+/// batch is filtered out.
```

Comment:
Now this function will always do filter work, so comments and the function name needs some modification:
```suggestion
/// Filter the `sequenced_record_batch` according to the `predicate`.
```

And the function name should be renamed to `filter_record_batch`.